xl/player/player.py

```diff
@@ -504,8 +571,31 @@ def engine_autoadvance_get_next_track(self, gapless=False):
             if self._auto_advance_delay != 0 or not self._gapless_enabled:
                 return
 
+        track = self._track
+        stop_at = track.get_tag_raw('__stopoffset') or 0
+        track_len = track.get_tag_raw('__length') or 0
+
+        if stop_at > track_len:
```

Comment:
This doesn't seem like the right place to implement this. As the documentation for the function says, "the (next) track may or may not actually get played."

Reply:
So it is. At least I didn't find a better place.
`engine_autoadvance_get_next_track` is called by `gst/engine.py::_autoadvance_track`, which is called when the current stream ends. In this moment the current track is still the last one (that with the stop offset in future).
I tried to move this to `engine_autoadvance_notify_next` but there the offset information is allready lost.